### DISCUSSION TITLE
Write a test

### DISCUSSION THREAD

**Discussion by QuangDom** [idaholab/moose#18952] [Q&A General] [answered]
Could someone point me to an example or documentation that I can learn about writing a test.
I would like to test the results by doing an integration over the whole simulation domain (a 2D surface); Or do doing a line integration over the outlet boundary and compare the integration results with a 'gold' values.

**Comment by GiudGiud** [comment, marked as answer]
Hello

The test system documentation is here https://mooseframework.inl.gov/python/TestHarness.html#!

You can have a look at moose/test/tests/kernels/simple_diffusion/ for an example of a test.
This test is an exodiff, so it compares the values of the variables everywhere. 

For a line integration, you may want to look at a CSVDiff test, as the postrprocessors are easily output to csv files. 
An example of that is in moose/test/tests/postprocessors/num_elems/

Guillaume
